### DISCUSSION TITLE
CropForeground behaviour

### DISCUSSION THREAD

**Discussion by schellenchris** [Project-MONAI/MONAI#5842] [Q&A] [answered]
I am croping my images to match a segmentation when I noticed my x-dimesnion isn't getting cropped somehow. I tested the CropForeground function:

image = np.array(
[[0, 0, 1],
[0, 1, 0],
[0, 0, 0]])
cropper = CropForeground(select_fn=lambda x: x > 0, margin=0)
print(cropper(image))

[[0, 1],
 [1, 0],
 [0, 0]]

Is this behaviour of CropForeground intended? Shouldn't it get cropped to

[[0, 1],
 [1, 0]]

Thanks for your help

**Comment by KumoLiu** [comment, marked as answer]
Hi @schellenchris, since transforms in MONAI assume that the data is channel-first, so you need to add the channel dimension to the data before passing it into the transforms. Could you please try `print(cropper(image[None]))`?

Hope it can help you, thanks!

**Reply by schellenchris** [reply]
Thank you! My x-dimension was in fact the first dimension hence it wasn't getting cropped cause it assumed it was the channel..... My bad
